### ISSUE TITLE
Cache type without "instance" causes exception when disabling the module through "Cache Management" in the backend

### ISSUE THREAD

**Issue by Serializator** [magento/magento2#26224]
### Preconditions (*)
1. Magento 2.3.3 CE

### Steps to reproduce (*)

1. Create a cache type using "cache.xml" without the "instance" argument
```xml
<config .../>
    <type name="exceptional_cache_type">
        <label>Exceptional Cache Type</label>
        <description>An exceptional cache type using a quantum computer</description>
    </type>
</config>
```
2. Go to "System > Cache Management" in the backend
3. Check the checkbox of the cache type "Exceptional Cache Type"
4. Select "Disable" and click "Submit"

### Expected result (*)
I'd expect the cache type to be successfully disabled. 

According to the XSD ("lib/internal/Magento/Framework/Cache/etc/cache.xsd") the "instance" argument is optional and not required.

### Actual result (*)
It throws an exception because it's trying to invoke `->clean(...)` on a null variable. 

```
Fatal error: Uncaught Error: Call to a member function clean() on null in /lib/internal/Magento/Framework/App/Cache/TypeList.php:194 Stack trace:
#0 /app/code/Magento/Backend/Controller/Adminhtml/Cache/MassDisable.php(62): Magento\Framework\App\Cache\TypeList->cleanType('invalid_cache_t...')
#1 /app/code/Magento/Backend/Controller/Adminhtml/Cache/MassDisable.php(41): Magento\Backend\Controller\Adminhtml\Cache\MassDisable->disableCache()
#2 /generated/code/Magento/Backend/Controller/Adminhtml/Cache/MassDisable/Interceptor.php(24): Magento\Backend\Controller\Adminhtml\Cache\MassDisable->execute()
#3 /lib/internal/Magento/Framework/App/Action/Action.php(107): Magento\Backend\Controller\Adminhtml\Cache\MassDisable\Interceptor->execute()
```

**Comment by eduard13** [comment]
Hi @Serializator, I was able to reproduce it, and working on a fix for it. Please note that I've updated the description a bit, by adjusting the path for the XSD file.
Thanks.

**Comment by rogyar** [comment]
Hi folks. I can confirm that this issue appears. But is there a use case for a cache definition without an implementation? I mean, maybe it's better to throw an exception to notify the developer that the parameter is missing than silently accept the "ghost" cache type?

**Comment by Serializator** [comment]
Hi @rogyar, I think there are two possible solutions to the problem and it depends on what the leading factor is. 

If you take the XSD as "source of truth" (or whatever you wanna call it, hahah) it shouldn't throw the exception. If you take the implementation as the leading factor the XSD should be updated and make the "instance" parameter required.

I talked a little with @eduard13 in the Magento Community Engineering Slack about why `Magento\Framework\App\Cache\TypeList` is using a different factory than `Magento\Framework\App\Cache\Frontend\Pool`.

`Magento\Framework\App\Cache\TypeList` uses `Magento\Framework\App\Cache\InstanceFactory`, which basically instantiates the given class using the object manager and validates that its actually of type `Magento\Framework\Cache\FrontendInterface`.

`Magento\Framework\App\Cache\Frontend\Pool` uses `Magento\Framework\App\Cache\Frontend\Factory`. Note that I'm making the assumption that both factories have the same purpose because they both return classes of type `Magento\Framework\App\Cache\FrontendInterface`. I'm also assuming that `$frontend['type']` on line 150 in `Magento\Framework\App\Cache\Frontend\Pool` is equivalent to the "instance" parameter.

A lot of assumptions, I know.

If `$frontend['type']` doesn't exist it uses `Magento\Framework\Cache\Core` as the type, negating the possible null exception that we have when using `Magento\Framework\App\Cache\InstanceFatory`.

**Comment by rogyar** [comment]
Hi @Serializator. Thanks very interesting observation and absolutely makes sense for me. Thank you for the detailed description. 

I believe we need to figure out why the `TypeList` uses a different factory and what is the best way to proceed in case if the frontend type is null. I will raise this question during an internal discussion.

**Comment by Serializator** [comment]
Hi @rogyar, any progress in regards to the internal discussion or not yet? Just checking out of curiosity 😉

**Comment by rogyar** [comment]
Still in progress. I will get back here as soon as I have green/red light on how to proceed with this one.

Thank you

**Comment by magento-engcom-team** [comment]
Hi @Serializator. Thank you for your report. 
The issue has been fixed in magento/magento2#27307 by @andrewbess in 2.4-develop branch
Related commit(s):
 - [97bd66e984895f12c8f17b7f401ebf352a28805c](https://github.com/magento/magento2/commit/97bd66e984895f12c8f17b7f401ebf352a28805c)
 - [34c2b6ab29f157ca5c228a27d69018d4c69c8df9](https://github.com/magento/magento2/commit/34c2b6ab29f157ca5c228a27d69018d4c69c8df9)
 - [c1662ffe291c04daaa362605e2c641fa6aa4b71c](https://github.com/magento/magento2/commit/c1662ffe291c04daaa362605e2c641fa6aa4b71c)

The fix will be available with the upcoming 2.4.0 release.
